packages/core/src/components/widgets/RadioWidget.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FocusEvent } from "react";
+import { WidgetProps } from "@rjsf/utils";
+
+/** The `RadioWidget` is a widget for rendering a radio group.
+ *  It is typically used with a string property constrained with enum options.
+ *
+ * @param props - The `WidgetProps` for this component
+ */
+function RadioWidget<T = any, F = any>({
+  options,
+  value,
+  required,
+  disabled,
+  readonly,
+  autofocus = false,
+  onBlur,
+  onFocus,
+  onChange,
+  id,
+}: WidgetProps<T, F>) {
+  // Generating a unique field name to identify this set of radio buttons
+  const name = Math.random().toString();
+  const { enumOptions, enumDisabled, inline } = options;
+  // checked={checked} has been moved above name={name}, As mentioned in #349;
+  // this is a temporary fix for radio button rendering bug in React, facebook/react#7630.
+  return (
+    <div className="field-radio-group" id={id}>
+      {Array.isArray(enumOptions) &&
+        enumOptions.map((option, i) => {
+          const checked = option.value === value;
+          const itemDisabled =
+            enumDisabled && enumDisabled.indexOf(option.value) != -1;
+          const disabledCls =
+            disabled || itemDisabled || readonly ? "disabled" : "";
+
+          const handleChange = () => {
+            if (onChange) {
+              return onChange(option.value);
+            }
+          };
+
+          const handleBlur = (event: FocusEvent<HTMLInputElement>) => {
+            if (onBlur) {
+              return onBlur(id, event.target.value);
+            }
+          };
+
+          const handleFocus = (event: FocusEvent<HTMLInputElement>) => {
+            if (onFocus) {
+              return onFocus(id, event.target.value);
+            }
+          };
```

Comment:
These two could also be moved outside of the map loop since they are not dependent on anything inside of the loop